doc/doxygen/headers/reordering.h

```diff
@@ -19,17 +19,68 @@
+ * detail of the library. Normal users should not need to use the features
+ * describd in this module: instead, classes like QProjector use orientation
+ * information to consistently compute values on faces and lines.
+ *
+ * <h2>Orientation of Lines</h2>
+ *
+ * A Triangulation is built not just of cells but also of lower-dimensional
+ * objects. In particular, a line in 2D is a face which may be shared by two
+ * cells, whereas in 3D a line may be shared by an arbitrary number of cells.
+ *
+ * Lines (i.e., faces in 2D) are implicitly defined by the vertex numbering of
+ * the cell on which they are defined (i.e., there is no separate line or face
+ * object). This information is encoded in various ReferenceCell functions. For
+ * example, the first face of a triangle with nodes {0, 1, 2} is {0, 1}.
+ * Similarly, the first face of a triangle with nodes {1, 0, 3} is {1, 0}.
+ * By itself, this would be inconsistent, since the same line would be defined
+ * in two diferent ways. To solve this problem deal.II also stores an
+ * <tt>unsigned char</tt> whose value encodes the orientation of the line. In
+ * this particular case the orientation of the {1, 0} line is
+ * ReferenceCell::default_combined_face_orientation() whereas the orientation of
+ * the {0, 1} line is ReferenceCell::reversed_combined_line_orientation().
+ *
+ * <h2>Orientation of Faces</h2>
+ *
+ * In deal.II, we express the orientation of an object with three booleans:
+ * orientation, rotate, and flip. The default values for these are true, false,
+ * and false. These are represented either as individual booleans or as a
+ * combined orientation packed into a single <tt>unsigned char</tt>: the
+ * combined format places orientation in the least significant bit, then rotate,
+ * then flip.
+ *
+ * For a quadrilateral, these values correspond to
+ * - *orientation* : `true` is the default orientation and `false` means
+ *   vertices 1 and 2 are swapped.
+ * - *rotation* : all vertices are rotated by 90 degrees clockwise.
+ * - *flip* : all vertices are rotated by 180 degrees clockwise.
+ *
+ * For a triangle, these values correspond to
+ * - *orientation* : `true` is the default orientation and `false` means
+ *   vertices 1 and 2 are swapped.
+ * - *rotation* : all vertices are rotated by 60 degrees clockwise.
+ * - *flip* : all vertices are rotated by 120 degrees clockwise.
+ *
+ * Here, 'clockwise' is relative to the vector defined by the cross
+ * product of two lines in their standard orientation (which, e.g., points
+ * into the hexahedron for face 0 but out of the hexahedron for face 1).
+ *
+ * For triangles, to enable indexing from the combined orientation, we do
+ * not consider flip-rotate or flip-orient-rotate as those cases are
+ * equivalent, respectively, to the identity operation or the orientation =
+ * `true` case as flip-rotate is equal to the identity operation. This
+ * choice ensures that the integer value of the combined orientation is in
+ * $[0, 5]$.
```

Comment:
Like before, I would omit the specific encoding and only say the following:
```suggestion
 * `true` case as flip-rotate is equal to the identity operation. As a consequence,
 * there are only six valid orientations for triangles as faces of hexahedra.
```